app/Http/Controllers/Profile/ProfileController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers\Profile;
+
+use Kris\LaravelFormBuilder\FormBuilderTrait;
+use Illuminate\Http\Request;
+use App\Member;
+use App\Forms\UpdateProfileForm;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use Auth;
+use JsValidator;
+use Session;
+
+class ProfileController extends Controller
+{
+    use FormBuilderTrait;
+	
+    public function show()
+    {
+		$user = Auth::user();
+		
+		if($user === null){
+			Session::put('alert-login', 'Please login to view/update your profile');
+			return redirect('login');
+		}
+		
+        $form = $this->form(UpdateProfileForm::class,[
+            'method' => 'POST',
+            'url' => route('member.profile.update'),
+        ]);
+        return view('members.profile.index', compact('form'))->with([
+            'validator' => JsValidator::make($this->profileRules()),
+        ]);
+    }
+	
+	public function save(Request $request)
+	{
+		$form = $this->form(UpdateProfileForm::class);
+        $form->validate($this->profileRules(), [
+            'nricformat' => 'NRIC checksum failed. Try checking it again.',
+            'dateformat' => 'Date should be a valid date of the format YYYY-MM-DD',
+        ]);
+        if (!$form->isValid()) {
+            return redirect()->back()->with('alert-warning', 'Error in form input!')->withErrors($form->getErrors())->withInput();
+        }
+		
+		$user = Auth::user();
+		// get the first member where the crsid matches and update the model
+		$person = Member::where('crsid', $user->crsid)
+			->first()
+			->update($request->all());
+		return redirect()->route('member.profile')->with('alert-success', 'Profile updated.');
+	}
+	
+	public function profileRules(){
+		$rules = [
+            'first_name' => 'required',
+            'last_name' => 'required',
+            'gender' => 'required',
+            'date_of_birth' => 'sometimes|required|dateformat:Y-m-d',
+            'email_other' => 'required|email',
+            'email_hermes' => 'sometimes|required|email',
```

Comment:
Try a regex ending with *cam.ac.uk, pretty sure department emails are subdomains of cam.ac.uk, e.g. cl.cam.ac.uk, phy.cam.ac.uk, etc.